java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+}
+```
+
+When the module is started, a timer needs to be created to send temperature data. We do not need to release any resources (including the timer) because our timer is a [daemon](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#setDaemon-boolean-).
+
+```java
+@Override
+public void start() {
+  new Timer(true).schedule(new TimerTask() {
+    private Random rand = new Random();
+
+    @Override
+    public void run() {
+      try {
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put("source", "bleTelemetry");
+        properties.put("macAddress", SimulatedBleModule.this.macAddress);
+
+        float temperature = this.rand.nextFloat() * 20 + 20;
+        byte[] content = Float.toString(temperature).getBytes();
+        
+        SimulatedBleModule.this.publish(new Message(content, properties));
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    }
+  }, 0, this.messagePeriod);
+}
+```
+
+### Converter Module
+
+| Input                    | Processor                              | Output                 | Source File            |
+| ------------------------ | -------------------------------------- | ---------------------- | ---------------------- |
+| Temperature data message | Parse and construct a new JSON message | Structure JSON message | `ConverterModule.java` |
+
+This module is a typical Azure IoT Edge module. It accepts temperature message from another module (a hardware module, or in this case our simulated BLE module); and then it normalizes the temperature message to a structured JSON message (including appending the message ID, setting the property of whether we need to trigger the temperature alert, and so on).
+
+```java
+@Override
+public void receive(Message message) {
+  try {
+    float temperature = Float.parseFloat(new String(message.getContent()));
+    Map<String, String> inputProperties = message.getProperties();
+
+    HashMap<String, String> properties = new HashMap<>();
+    properties.put("source", inputProperties.get("source"));
+    properties.put("macAddress", inputProperties.get("macAddress"));
+    properties.put("temperatureAlert", temperature > 30 ? "true" : "false");
+
+    String content = String.format(
+      "{ \"deviceId\": \"Intel NUC Gateway\", \"messageId\": %d, \"temperature\": %f }",
+      ++this.messageCount, temperature);
+
+    this.publish(new Message(content.getBytes(), properties));
+  } catch (Exception ex) {
+    ex.printStackTrace();
+  }
+}
+```
+
+### Printer Module
+
+| Input                          | Processor | Output                     | Source File          |
+| ------------------------------ | --------- | -------------------------- | -------------------- |
+| Any message from other modules | N/A       | Log the message to console | `PrinterModule.java` |
+
+This module is the simplest one among all the modules we create. Let's just show the code without explaining it.
```

Comment:
This is a very simple, self-explanatory, module which outputs the received messages to the terminal window.